File: pe/visualstudio/PE.FSharp/Problem0006.fs
namespace PE.FSharp.Problem6

open Microsoft.FSharp.Core 
open Microsoft.FSharp.Core.Operators 

module SumSquareDifference =
    
    let sumOfSquares (list: int64 list) = 
        list
        |> List.map (fun element -> element * element)
        |> List.sum

    let squareOfSum (list: int64 list) = 
        list
        |> List.sum 
        |> (fun sumOfList -> sumOfList * sumOfList) 


    let differenceBetweenSumOfSquarsAndSquareOfSum (list: int64 list) =
        (squareOfSum list) - (sumOfSquares list) 

File: pe/nonvisualstudio/haskell/OldWork/Implementation/Problem0006.hs
module Problem0006 where

    sumOfSquares max = sum (map (^2) [1..max])

    squareOfSum max = (sum [1..max]) ^ 2

    differenceBetweenSumOfSquarsAndSquareOfSum max = (squareOfSum max) - (sumOfSquares max)
